pkg/routeconfig/routeconfig.go

```diff
@@ -0,0 +1,144 @@
+package routeconfig
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"regexp"
+	"text/template"
+
+	"github.com/flynn/flynn/controller/api"
+	router "github.com/flynn/flynn/router/types"
+	"github.com/stripe/skycfg"
+)
+
+// Load loads app routes from the given route config
+func Load(routeConfig io.Reader) ([]*api.AppRoutes, error) {
+	// skycfg wants to read from a single file, so create a temp file
+	// to hold both the main and the supplied config
+	tmp, err := ioutil.TempFile("", "flynn-route-config-")
+	if err != nil {
+		return nil, err
+	}
+	defer os.Remove(tmp.Name())
+	defer tmp.Close()
+
+	// write the main config followed by the route config to the tmp file
+	if _, err := io.WriteString(tmp, configMain); err != nil {
+		return nil, fmt.Errorf("error creating config file: %s", err)
+	}
+	if _, err := io.Copy(tmp, routeConfig); err != nil {
+		return nil, fmt.Errorf("error creating config file: %s", err)
+	}
+
+	// flush the config to disk by closing the tmp file
+	if err := tmp.Close(); err != nil {
+		return nil, fmt.Errorf("error creating config file: %s", err)
+	}
+
+	// load the config using skycfg
+	ctx := context.Background()
+	config, err := skycfg.Load(ctx, tmp.Name())
```

Comment:
Use a `skycfg.FileReader` that reads from memory instead of writing to a tempfile.

Reply:
Done in 151ec5a1 (and refactored in 77cab2b).